tests/api/test_record_delete.py

```diff
@@ -52,3 +52,16 @@ def test_delete_internal_location(client, users, json_headers, testdata):
     )
     res = client.delete(url, headers=json_headers)
     assert res.status_code == 204
+
+
+def test_delete_keyword(client, users, json_headers, testdata):
+    """Test DELETE existing keyword."""
+    login_user_via_session(
+        client,
+        email=User.query.get(users["admin"].id).email
+    )
+
+    keyword_pid = 'keyid-1'
+    url = url_for('invenio_records_rest.keyid_item', pid_value=keyword_pid)
+    res = client.delete(url, headers=json_headers)
```

Comment:
If you put `deny_all`, then the test will fail. Just remove it IMHO.
You can test the delete method if you want.

```
k1 = Keywords.create({...})
k.delete()
```
But it would make more sense to test the deletion fail instead of the success case...

Reply:
Yes the crucial part is to test prevent of deletion for the record with references